[MRG] MAINT: add fixture for init and clean-up with matplotlib (#13708)
scikit-learn/scikit-learn: doc/developers/contributing.rst

We expect code coverage of new features to be at least around 90%.

For guidelines on how to use ``pytest`` efficiently, see the
:ref:`pytest_tips`.

> Available test fixtures in scikit-learn

Reviewer: I would suggest to call this section

> Writing matplotlib related tests

instead, and change the writing accordingly. Again, it's weird to mention multiple fixtures when we only have one.

.......................................

Test fixtures ensure that a set of tests will be executing with the appropriate
initialization and cleanup. The scikit-learn test suite implement a fixture
which can be used with ``matplotlib``.

> `pyplot`

Reviewer: double backticks (here and below)

> `plt`

Reviewer: should be pyplot now? (here and in other places)

    The `plt` fixture should be used when a test function is dealing with
    `matplotlib`. `matplotlib` is a soft dependency and is not required. This

>     fixture is charge of skipping the tests if `matplotlib` is not installed.

Reviewer: ```suggestion
    fixture is in charge of skipping the tests if `matplotlib` is not installed.
```

    installed. In addition, figures created during the tests will be
    automatically closed once the test function has been executed.

> To use one of these fixtures in a test function, one needs to pass it as an

Reviewer: ```suggestion
To use this fixture in a test function, one needs to pass it as an
```

> To use one of these fixtures in a test function, one needs to pass it as an

Reviewer: We only have one fixture documented here, and yet the whole section suggests there are more than one. It's not that confusing, but it's a bit strange.

argument::

    def test_requiring_mpl_fixture(plt):

>         # you can now use safely matplotlib

Reviewer: ```suggestion
        # you can now use matplotlib safely
```
